lib/sierra-models/item.js

```diff
@@ -31,6 +39,55 @@ class SierraItem extends SierraBase {
     return type
   }
 
+  getIsResearchWithRationale () {
```

Comment:
I think it would be useful to summarize the checks that are being made here.

Reply:
I refactored a bit and added a bunch of documentation.

Reply:
🤩 looks lovely!